mlx/backend/metal/kernels/binary.metal

```diff
@@ -15,10 +15,33 @@ struct Divide {
 };
 
 struct Remainder {
-  template <typename T> T operator()(T x, T y) { return x % y; }
-  template <> float operator()(float x, float y) { return fmod(x, y); }
-  template <> half operator()(half x, half y) { return fmod(x, y); }
-  template <> bfloat16_t operator()(bfloat16_t x, bfloat16_t y) { return fmod(x, y); }
+  template <typename T> T operator()(T x, T y) {
+    auto r = x % y;
+    if (r < 0 && y > 0)
+      r+= y;
+    return r; 
+  }
+  template <> complex64_t operator()(complex64_t x, complex64_t y) {
+    return x % y; 
+  }
+  template <> float operator()(float x, float y) {
+    float r = fmod(x, y);
+    if (r < 0 && y >0)
+      r+= y;
```

Comment:
Spaces after `r` and `y >`. Possibly we also need an unsigned numbers specialization that doesn't do the check. Otoh the compiler should be able to optimize that away so perhaps we are good.

Reply:
I added specialization for eval_cpu and eval, but i am not sure how to do it in metal.